tensorflow/compiler/xla/tests/dot_operation_test.cc

```diff
@@ -734,6 +737,340 @@ XLA_TYPED_TEST(DotOperationTestWithCublasLt_F16F32F64CF64,
       &builder, expected, {x_data.get(), y_data.get()}, this->error_spec_);
 }
 
+template <typename T>
+class DotOperationTestWithCublasLt_F8 : public DotOperationTest {
+ public:
+  DotOperationTestWithCublasLt_F8() {
+    execution_options_.mutable_debug_options()->set_xla_gpu_enable_cublaslt(
+        true);
+  }
+};
+TYPED_TEST_CASE(DotOperationTestWithCublasLt_F8, TypesF8);
+
+XLA_TYPED_TEST(DotOperationTestWithCublasLt_F8, ScaledABUnscaledDF8) {
+  using T = TypeParam;
+
+  XlaBuilder builder(this->TestName());
+  XlaOp a =
+      Parameter(&builder, 0, ShapeUtil::MakeShapeWithType<T>({16, 16}), "A");
+  XlaOp b =
+      Parameter(&builder, 1, ShapeUtil::MakeShapeWithType<T>({16, 16}), "B");
+  XlaOp a_scale = Parameter(&builder, 2,
+                            ShapeUtil::MakeShapeWithType<float>({}), "A scale");
+  XlaOp b_scale = Parameter(&builder, 3,
+                            ShapeUtil::MakeShapeWithType<float>({}), "B scale");
+
+  DotDimensionNumbers dnums;
+  dnums.add_lhs_contracting_dimensions(1);
+  dnums.add_rhs_contracting_dimensions(0);
+
+  XlaOp a_f32 = ConvertElementType(a, F32);
+  XlaOp b_f32 = ConvertElementType(b, F32);
+
+  XlaOp a_scale_bcast = Broadcast(a_scale, {16, 16});
+  XlaOp a_scaled_f32 = Mul(a_f32, a_scale_bcast);
+  XlaOp b_scale_bcast = Broadcast(b_scale, {16, 16});
+  XlaOp b_scaled_f32 = Mul(b_f32, b_scale_bcast);
+
+  XlaOp d_f32 = DotGeneral(a_scaled_f32, b_scaled_f32, dnums);
+
+  auto a_data = this->client_
```

Comment:
It is difficult to verify the test is correct. Can you instead create smaller matrices, such as 3x3 matrices, then pad them to be size 16x16?

Also making the scales whole numbers would make it easier to manually verify the test is correct.

Reply:
I considered this when implementing the test but decided against it. Having only a small block of non-zero elements would mean that most (more than 95% in the case of 3 x 3) of the entries are zeros, which is not ideal for verifying memory allocations and reads/writes. This test typically also runs on CPU which can be seen as an additional verification.

Reply:
We could limit the non-zero entries to two 2 x 2 blocks at the beginning and the end of the diagonals of A and B. I think that would resolve both issues.

Reply:
Having two 2x2 blocks SGTM. Alternatively, you can generate random input data and test the CPU and GPU give the same results, or that GPU F8 and GPU f32 give the same results.